### DISCUSSION TITLE
More intuitive shell interface -- feature idea

### DISCUSSION THREAD

**Discussion by DarrellDuane** [fish-shell/fish-shell#10065] [Ideas]
At present, in all shells that I know of, if I attempt to move files to a directory that doesn't exist, I get an error message. 

What if, in addition to an error message, the shell asked "do you want to create this directory and move the files (y/n)"

Would Fish welcome this enhacement ?  Is ther another way we should go  about it?  Or shall I consider using ai-shell instead.  

```
user@server_name ~/camera> mv 2023-10-18\ 08*  /home/user/Downloads/car
mv: target '/home/user/Downloads/car' is not a directory

```

**Comment by faho** [comment]
This isn't doable in fish's model. `mv` is an external command that does *whatever it does*, and fish has no idea that the error was that the directory does not exist. There is no usable channel to differentiate e.g. permission problems or "it's a file, not a directory" from "doesn't exist".

You *could* make a function that creates the directory if it doesn't exist, or find an alternative to `mv` that auto-creates any directories. We're not really interested in doing that by default because it would be contrary to what people think `mv` does.
